spring-cloud-gcp-data-datastore/src/main/java/org/springframework/cloud/gcp/data/datastore/core/DatastoreTemplate.java

```diff
@@ -228,12 +231,13 @@ private Entity convertToEntityForSave(Object entity) {
 		DatastorePersistentEntity datastorePersistentEntity = this.datastoreMappingContext
 				.getPersistentEntity(entityClass);
 
-		entities.forEachRemaining(entity -> convertEntityResolveDescendants(entityClass,
+		entities.forEachRemaining(entity -> convertEntityResolveDescendantsAndReferences(
+				entityClass,
 				results, datastorePersistentEntity, entity));
 		return results;
 	}
 
-	private <T> void convertEntityResolveDescendants(Class<T> entityClass,
+	private <T> void convertEntityResolveDescendantsAndReferences(Class<T> entityClass,
 			List<T> results, DatastorePersistentEntity datastorePersistentEntity,
 			BaseEntity entity) {
 		T convertedObject = this.datastoreEntityConverter.read(entityClass, entity);
```

Comment:
I would break this long method up into two private methods - one for descendants and one for references.

Reply:
@meltsufin change made. PTAL